### PR TITLE
Add chart section for personality type voting.

### PR DESCRIPTION
**Pull request by keelywan**
## After
chart

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/PersonalityDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+public class PersonalityDataServlet extends HttpServlet {
+
+  private Map<String, Integer> personalityTypeVotes = new HashMap<>();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    Gson gson = new Gson();
+    String json = gson.toJson(personalityTypeVotes);
+    response.getWriter().println(json);
```
Comment (nickKelman) [inline portfolio/src/main/java/com/google/sps/servlets/PersonalityDataServlet.java:38]: I assume you don't need this.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -161,6 +163,47 @@ <h2>~ hiking ~</h2>
...
+        <p>
+          <ul>
+            <li>Where you focus your attention - extraversion (E) or introversion(I)</li>
+            <li>The way you take in information - sensing (S) or intuition (N)</li>
+            <li>How you make decisions - thinking (T) or feeling (F)</li>
+            <li>How you deal with the world - judging (J) or perceiving (P)</li>
+          </ul>
+        </p>
+        <p class="section-descr">
+          Take the test <a href="https://www.16personalities.com/free-personality-test">here</a>!
```
Comment (nickKelman) [inline portfolio/src/main/webapp/index.html:181]: I remember there being so many questions to this test, I can't remember what type I am

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -221,4 +221,34 @@ function initializePage() {
...
+    const data = new google.visualization.DataTable();
+    data.addColumn('string', 'Personality Type');
+    data.addColumn('number', 'Votes');
+    data.addColumn({type:'string', role:'style'});
+    Object.keys(personalityVotes).forEach((type) => {
+      data.addRow([type, personalityVotes[type], 'color:#5c7fe0']);
+    });
+
+    const options = {
+      'title': 'Personality Types',
```
Comment (nickKelman) [inline portfolio/src/main/webapp/script.js:244]: I would be consistent about declaring keys with quotes or without within the same object.

**Review by nickKelman — APPROVED** [review]
Very cool

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -161,6 +163,47 @@ <h2>~ hiking ~</h2>
...
+        </p>
+        <div id="chart-container"></div>
+        <form action="/personality-data" method="POST">
+        <select name="type">
+          <option value="INTJ">Architect (INTJ)</option>
+          <option value="INTP">Logician (INTP)</option>
+          <option value="ENTJ">Commander (ENTJ)</option>
+          <option value="ENTP">Debater (ENTP)</option>
+          <option value="INFJ">Advocate (INFJ)</option>
```
Comment (cftorres) [inline portfolio/src/main/webapp/index.html:190]: I'm an Advocate :D Which one did you say you were?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -221,4 +221,34 @@ function initializePage() {
...
+ * Fetches personality data and uses it to create a chart.
+ */
+function drawChart() {
+  fetch('/personality-data').then(response => response.json())
+  .then((personalityVotes) => {
+    const data = new google.visualization.DataTable();
+    data.addColumn('string', 'Personality Type');
+    data.addColumn('number', 'Votes');
+    data.addColumn({type:'string', role:'style'});
+    Object.keys(personalityVotes).forEach((type) => {
```
Comment (cftorres) [inline portfolio/src/main/webapp/script.js:239]: you can also use Object.entries to get [key, value]

**Review by cftorres — APPROVED** [review]

File: portfolio/src/main/webapp/index.html (hunk 4 above)
Comment (keelywan) [reply portfolio/src/main/webapp/index.html:190]: I used to be a Defender, but I recently took it again and now I'm a Virtuoso.

File: portfolio/src/main/java/com/google/sps/servlets/PersonalityDataServlet.java (hunk 1 above)
Comment (keelywan) [reply portfolio/src/main/java/com/google/sps/servlets/PersonalityDataServlet.java:38]: I think I still need it in the drawChart() function to fetch all the votes and display it in the chart.

**Pull request merged**